src/blocking/queue.rs

```diff
@@ -198,18 +199,21 @@ impl Queue {
             .ok()
     }
 
-    pub fn fetch_periodic_tasks(&self, error_margin_seconds: i64) -> Option<Vec<PeriodicTask>> {
+    pub fn fetch_periodic_tasks(
+        &self,
+        error_margin_seconds: StdDuration,
+    ) -> Option<Vec<PeriodicTask>> {
         Self::fetch_periodic_tasks_query(&self.connection, error_margin_seconds)
     }
 
     pub fn fetch_periodic_tasks_query(
         connection: &PgConnection,
-        error_margin_seconds: i64,
+        error_margin_seconds: StdDuration,
     ) -> Option<Vec<PeriodicTask>> {
         let current_time = Self::current_time();
 
-        let low_limit = current_time - Duration::seconds(error_margin_seconds);
-        let high_limit = current_time + Duration::seconds(error_margin_seconds);
+        let low_limit = current_time - Duration::from_std(error_margin_seconds).ok()?;
+        let high_limit = current_time + Duration::from_std(error_margin_seconds).ok()?;
```

Comment:
I don't think it's a good approach to handle errors like this

Reply:
[Error management](https://github.com/ayrat555/fang/pull/67/commits/b3fb2ab6b20328a26099ca8cce2f9f79d38cfc84)